domenetjenester/behandling-revurdering/src/main/java/no/nav/foreldrepenger/behandling/revurdering/satsregulering/EngangsstønadReguleringTask.java

```diff
@@ -0,0 +1,110 @@
+
+    private static final Logger LOG = LoggerFactory.getLogger(EngangsstønadReguleringTask.class);
+
+    private final RevurderingTjeneste revurderingTjeneste;
+    private final LegacyESBeregningRepository esBeregningRepository;
+    private final BehandlingProsesseringTjeneste behandlingProsesseringTjeneste;
+    private final PersoninfoAdapter personinfoAdapter;
+    private final BehandlingRepository behandlingRepository;
+    private final FamilieHendelseTjeneste familieHendelseTjeneste;
+    private final BehandlendeEnhetTjeneste behandlendeEnhetTjeneste;
+
+    @Inject
+    public EngangsstønadReguleringTask(BehandlingRepositoryProvider repositoryProvider,
+                                       FamilieHendelseTjeneste familieHendelseTjeneste,
+                                       PersoninfoAdapter personinfoAdapter,
+                                       BehandlendeEnhetTjeneste behandlendeEnhetTjeneste,
+                                       LegacyESBeregningRepository esBeregningRepository,
+                                       BehandlingProsesseringTjeneste behandlingProsesseringTjeneste,
+                                       @FagsakYtelseTypeRef(FagsakYtelseType.ENGANGSTØNAD) RevurderingTjeneste revurderingTjeneste) {
+        super(repositoryProvider.getFagsakLåsRepository(), repositoryProvider.getBehandlingLåsRepository());
+        this.familieHendelseTjeneste = familieHendelseTjeneste;
+        this.personinfoAdapter = personinfoAdapter;
+        this.behandlingRepository = repositoryProvider.getBehandlingRepository();
+        this.behandlendeEnhetTjeneste = behandlendeEnhetTjeneste;
+        this.revurderingTjeneste = revurderingTjeneste;
+        this.esBeregningRepository = esBeregningRepository;
+        this.behandlingProsesseringTjeneste = behandlingProsesseringTjeneste;
+    }
+
+    @Override
+    protected void prosesser(ProsessTaskData prosessTaskData, Long fagsakId, Long behandlingId) {
+        LOG.info("ESregulering fagsak med fagsakId = {}", fagsakId);
+
+        var behandling = behandlingRepository.hentBehandling(behandlingId);
+
+        if (!FagsakYtelseType.ENGANGSTØNAD.equals(behandling.getFagsakYtelseType()) ||
+            behandlingRepository.harÅpenOrdinærYtelseBehandlingerForFagsakId(fagsakId)) {
+            return;
+        }
+
+        familieHendelseTjeneste.finnAggregat(behandling.getId())
+            .flatMap(g -> utledRevurderingsbehovMedÅrsak(behandling, g))
+            .ifPresent(årsak -> {
+                var enhet = behandlendeEnhetTjeneste.finnBehandlendeEnhetFor(behandling.getFagsak());
+                opprettRevurdering(behandling, årsak, enhet);
+            });
+    }
+
+    public Optional<BehandlingÅrsakType> utledRevurderingsbehovMedÅrsak(Behandling behandling, FamilieHendelseGrunnlagEntitet grunnlag) {
+        var bekreftetBarn = grunnlag.getGjeldendeBekreftetVersjon()
+            .filter(fh -> FamilieHendelseType.FØDSEL.equals(fh.getType()) || fh.getGjelderAdopsjon())
```

Comment:
Hmm, hvorfor er type termin ekskludert her?

Reply:
Regner med at det er pga de som er beregnet på termin blir uansett revurdert ved fødsel.

Reply:
Engansstønad blir ikke revurdert ved fødselshendelse - slik som FP - bare revurdering hvis det blir ny sats pga dato.
Derfor må vi spørre PDL om fødselsdato i disse tilfellene (de har fh.overstyrt = bekreftet termindato). 
Har laget egen TFP for å utvide etterkontroll-tabellen med antall barn og fødselsdato. Marte må tygge litt på den,